src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala

```diff
@@ -48,32 +46,121 @@ final case class EServicesApiServiceImpl(
+      agreement = agreement.map(a => CompactAgreement(id = a.id, state = a.state.toApi)),
+      canSubscribe = certifiedAttributesSatisfied(
+        eService.attributes.toManagement,
+        requesterTenant.attributes.mapFilter(_.certified)
+      ),
+      isSubscribed = agreement.exists(a => SUBSCRIBED_AGREEMENT_STATES.contains(a.state)),
+      isMine = eService.producerId.toString == requesterId,
+      mail = requesterTenant.mails.headOption.map(_.toApi)
+    )
 
     onComplete(result) {
       handleError(s"Error retrieving descriptor $descriptorId of eservice $eserviceId") orElse {
         case Failure(x: EServiceDescriptorNotFound) =>
           logger.warn(x.getMessage)
-          getEserviceDescriptor404(problemOf(StatusCodes.NotFound, x))
-        case Success(descriptor)                    => getEserviceDescriptor200(descriptor)
+          getCatalogEServiceDescriptor404(problemOf(StatusCodes.NotFound, x))
+        case Success(descriptor)                    => getCatalogEServiceDescriptor200(descriptor)
       }
     }
   }
 
+  private def extractIdsFromAttributes(attributes: CatalogProcess.Attributes): Seq[UUID] =
+    attributes.certified.flatMap(extractIdsFromAttribute) ++
+      attributes.declared.flatMap(extractIdsFromAttribute) ++
+      attributes.verified.flatMap(extractIdsFromAttribute)
+
+  private def extractIdsFromAttribute(attribute: CatalogProcess.Attribute): Seq[UUID] = {
+    val fromSingle: Seq[UUID] = attribute.single.toSeq.map(_.id)
+    val fromGroup: Seq[UUID]  = attribute.group.toSeq.flatMap(_.map(_.id))
+
+    fromSingle ++ fromGroup
+  }
+
+  private def convertToApiAttributes(
+    currentAttributes: CatalogProcess.Attributes,
+    attributes: Seq[AttributeManagement.Attribute]
+  ): EServiceAttributes = {
+    val attributeNames: Map[UUID, AttributeDetails] =
+      attributes.map(attr => attr.id -> AttributeDetails(attr.name, attr.description)).toMap
+
+    EServiceAttributes(
+      certified = currentAttributes.certified.map(convertToApiAttribute(attributeNames)),
+      declared = currentAttributes.declared.map(convertToApiAttribute(attributeNames)),
+      verified = currentAttributes.verified.map(convertToApiAttribute(attributeNames))
+    )
+  }
+
+  private def convertToApiAttribute(
+    attributeNames: Map[UUID, AttributeDetails]
+  )(attribute: CatalogProcess.Attribute): EServiceAttribute = EServiceAttribute(
+    single = attribute.single.map(convertToApiAttributeValue(attributeNames)),
+    group = attribute.group.map(values => values.map(convertToApiAttributeValue(attributeNames)))
+  )
+
+  private def convertToApiAttributeValue(
+    attributeNames: Map[UUID, AttributeDetails]
+  )(value: CatalogProcess.AttributeValue) = EServiceAttributeValue(
+    id = value.id,
+    // TODO how to manage this case? Raise an error/Default/Flat option values
+    // TODO for now default value "Unknown"
+    name = attributeNames.get(value.id).map(_.name).getOrElse("Unknown"),
+    // TODO same here
```

Comment:
These TODOs come from catalog-process, but the problem still persists.
Any suggestions?

Reply:
Maybe it's worth discussing in a meet

Reply:
are there attributes without name? is it feasible to convert it to mandatory? maybe just at api level and not persistence?

Reply:
The problem here comes from accessing the map via .get which returns an Option.

So we have two possible solutions:
- use a default value ("Unknown")
- raise an API error (probably a 500)

I think it's almost impossible to have a case where the map doesn't contain an attribute because that map comes from the attribute registry and therefore all the attributes set in an eservice are fetched from the registry itself.

Probably raising here an exception could be the right solution.
What do you think?

Reply:
sorry you're right, I misread.
I agree, probably raising an error here is better because something went unexpectedly wrong and we should be notified asap